packages/smooth_app/lib/widgets/tab_navigator.dart

```diff
@@ -23,8 +23,7 @@ class TabNavigator extends StatelessWidget {
     } else if (tabItem == BottomNavigationTab.History) {
       child = const HistoryPage();
     } else if (tabItem == BottomNavigationTab.Scan) {
-      // The ScanPage doesn't use the here build Navigator, so that it can update when its offstage status changes
-      return ScanPage(offstage: offstage, navigatorKey: navigatorKey);
+      child = const ScanPage();
```

Comment:
A couple of suggestions:
1. `final Widget child;` instead of `late final Widget child;` - we won't be "late", we are 100% confident that the variable will be populated
2. using `switch` with `enum`s is nicer, because `dart` complains when not all the `enum`s are in a `case`, and anyway it's easier to read. Of course here we're talking about only 3 values, with a low probability to grow...
```dart
final Widget child;
switch(tabItem) {
  case BottomNavigationTab.Profile:
    child = const UserPreferencesPage();
    break;
  case BottomNavigationTab.History:
    child = const HistoryPage();
    break;
  case BottomNavigationTab.Scan:
    child = const ScanPage();
    break;
}
```

Reply:
Sounds good